olive/model/handler/mixin/dummy_inputs.py

```diff
@@ -33,19 +34,15 @@ def get_dummy_dataloader_from_io_config(self):
             )
         return dataloader
 
-    def get_dummy_dataloader_from_hf_config(self):
+    def _get_dummy_dataloader_from_data_config(self, data_config: DataConfig):
         dataloader = None
-        if self.hf_config and self.hf_config.model_name and self.hf_config.task and self.hf_config.dataset:
+        if data_config:
             # need both model_name and task to get dummy inputs
-            logger.debug("Using hf_config.dataset to get dummy dataloader")
-            dataloader = data_config_template.huggingface_data_config_template(
-                self.hf_config.model_name,
-                self.hf_config.task,
-                **self.hf_config.dataset,
-            ).to_data_container()
+            logger.debug("Using data_config to build dummy dataloader")
+            dataloader = data_config.to_data_container()
         return dataloader
 
-    def get_dummy_inputs(self, filter_hook=None, filter_hook_kwargs=None):
+    def get_dummy_inputs(self, data_config=None, filter_hook=None, filter_hook_kwargs=None):
```

Comment:
is there a use case for the `data_configs` option? I think it's safe to remove this code path entirely. We could just remove the `get_dummy_dataloader_from_hf_config` entirely.

Reply:
Done!